src/linker/Linker/LinkContext.cs

```diff
@@ -605,6 +605,12 @@ public bool IsWarningSuppressed (int warningCode, MessageOrigin origin)
 
 			return Suppressions.IsSuppressed (warningCode, origin, out _);
 		}
+
+		public static WarnVersion GetWarningVersion (int code)
+		{
+			// TODO: Once we add warnings beyond .NET5, give them a new WarnVersion
+			return WarnVersion.ILLink5;
```

Comment:
If the code is in custom steps range we should return the null version - basically always show the warning (steps should specify warning version via the API)

Reply:
What is the custom step range? As far as I can tell we only expose CreateWarningMessage to custom steps, and we don't check for conflicts with built-in warnings - I thought this is what you meant to do in the PR you mentioned you were working on.

Reply:
https://github.com/mono/linker/blob/master/docs/error-codes.md#illink-errors-and-warnings specifically:
> The known codes are in the range 1000 to 6000. Custom steps should avoid using this range not to collide with existing or future error and warning codes.

Reply:
I see, then we should change the check in CreateWarningMessage. I'm hesitant to do that here since I don't know whether there are already steps relying on the current behavior - I'll file an issue about it.

Reply:
https://github.com/mono/linker/issues/1384